### PR TITLE
Enhance serverless' git repository auth support

### PR DESCRIPTION
**Pull request by tgorgol**
**Description**

Changes proposed in this pull request:

- Enhance serverless' git repository auth support

**Related issue(s)**
https://github.com/kyma-project/kyma/issues/8959

### CODE REVIEW

File: components/function-controller/deploy/jobinit/Dockerfile [hunk 1]
```diff
@@ -25,16 +25,15 @@ RUN go build -ldflags "-s -w" -a -o jobinit cmd/jobinit/main.go \
 && mv ./jobinit /app/jobinit \
 && if [ -f ${BASE_APP_DIR}/licenses ]; then mv ${BASE_APP_DIR}/licenses /app/licenses; fi
 
-# get latest CA certs
-FROM alpine:latest as certs
-RUN apk --update add ca-certificates
-
 # result container
-FROM scratch
+FROM alpine:3.11
```
Comment (m00g3n) [inline components/function-controller/deploy/jobinit/Dockerfile:29]: The alpine 3.12 is, it would be good to bump the version

Comment (pPrecel) [reply components/function-controller/deploy/jobinit/Dockerfile:29]: Why is `alpine` better than `scratch`?

File: components/function-controller/config/crd/bases/serverless.kyma-project.io_gitrepositories.yaml [hunk 2]
```diff
@@ -54,7 +54,6 @@ spec:
                   description: Type is the type of authentication
                   enum:
                   - basic
-                  - token
```
Comment (pPrecel) [inline components/function-controller/config/crd/bases/serverless.kyma-project.io_gitrepositories.yaml:57]: Why did you remove the `token` option? Is it not one of supported options?

Comment (tgorgol) [reply components/function-controller/config/crd/bases/serverless.kyma-project.io_gitrepositories.yaml:57]: Custom OAuth implementations differ between popular git providers, I think it's better to add a note to documentation mentioning the possibility of using token auth in a way described by the specific provider.

File: components/function-controller/deploy/jobinit/Dockerfile (hunk 1 above)
Comment (tgorgol) [reply components/function-controller/deploy/jobinit/Dockerfile:29]: We needed ssh in the container.

File: components/function-controller/config/crd/bases/serverless.kyma-project.io_gitrepositories.yaml (hunk 2 above)
Comment (pPrecel) [reply components/function-controller/config/crd/bases/serverless.kyma-project.io_gitrepositories.yaml:57]: Good point. Thanks!
